### ISSUE TITLE
[Bug] NPC Ethan/Lyra's (HGSS) Pokemon is random for unknown reason.

### ISSUE THREAD

**Issue by HappyFaceSpider** [Ajarmar/universal-pokemon-randomizer-zx#752]
**Describe the bug**
In HGSS the 'Best Friend' character (i.e. Ethan if player is girl and Lyra if player is boy) should have a Marill. However after I used the Randomizer Ethan/Lyra's pokemon got randomized too (Umbreon in one attempt, Grotle in another).  However the only setting i picked in the Randomizer was "Change Impossible Evolutions", so I don't think any NPC's pokemon should've been randomised. Is it a bug or did I simply do something wrong? Or is this normal? I'm worried that it means other NPC's pokemon also got randomised.

**To Reproduce**
1.  Load Pokemon SoulSilver ROM into Randomizer
2. Check "Change Impossible Evolutions"
3. Save new ROM
4. Launch new ROM

**Expected behavior**
I expected Ethan/Lyra to still have Marill and only trade evolutions of some pokemon to be changed. 

**Screenshots**
![EthanHasGrotle](https://github.com/Ajarmar/universal-pokemon-randomizer-zx/assets/73723705/b9c6d89d-129e-4cdf-86fa-32c0b8c72069)

I'm attaching log from the Randomizer.
[Pokemon - SoulSilver MYMOD.nds.log](https://github.com/Ajarmar/universal-pokemon-randomizer-zx/files/15447785/Pokemon.-.SoulSilver.MYMOD.nds.log)

**System (please complete the following information):**
 - Operating System: Windows 11 (but I tried it also on Archlinux and same thing happened)
 - Java Version: Version 8 Update 411 64-bit
 - Randomizer Version: v4.6.0.

**Additional context**
So, this is my first time using the randomizer, so it's possible I simply f*cked something up.

**Comment by MogenarZ** [comment]
I encountered this same issue - similarly, I only checked "Change Impossible Evolutions".

**Comment by Xero-IX** [comment]
Just wanted to add that I encountered the same issue with the same settings. I tried to adjust the settings a second time but got the same thing: first time was Blastoise, second was Girafarig. Did you notice any other problems with the playthrough or should I just run it as is?

**Comment by Fellshadow** [comment]
this isn't a bug, the randomizer deliberately always randomizes ethan/lyra's pokemon to indicate that the rom has gone through the randomizer program
In other games like FRLG it's the pokemon the professor throws out in the intro of the game that always gets randomized, i'm not sure why they decided to use ethan/lyra's pokemon here for HGSS instead of the pokemon oak throws out in the intro but it's a deliberate choice rather than a bug

**Comment by bellonarius** [comment]
Is it possible to remove this particular randomizing? Imo it would be better to have prof. Oak show us a different Pokemon, because it is kinda weird to see Lyra or Ethan walking around with a Milotic or a Slaking at the start and other encounters

**Comment by Syddraf** [comment]
Would like to second the request to remove this. Seems like an odd choice to force Ethan's / Lyra's Pokémon being randomised on people.

**Comment by Maximum-beans** [comment]
I've finally found people who've encountered the same thing! I'd also like to voice support for changing this. It could have easily been done to the Marill that Oak shows in the intro. An odd choice, I agree.

**Comment by faizjamil** [comment]
Yeah I agree this is an odd choice, but that might be cause I lack a reason for this behavior. Granted I only use this program for non-randomizer related patches. Mainly to patch out the impossible evolutions. If this isn't the right patcher to use for that then I'd like to know where I can get a patch and patcher for dealing with the trading evolutions. Only workaround I've found is to use an older version of this randomizer program

**Comment by MogenarZ** [comment]
> Yeah I agree this is an odd choice, but that might be cause I lack a reason for this behavior. Granted I only use this program for non-randomizer related patches. Mainly to patch out the impossible evolutions. If this isn't the right patcher to use for that then I'd like to know where I can get a patch and patcher for dealing with the trading evolutions. Only workaround I've found is to use an older version of this randomizer program

I ended up using the DSPRE, which has a fairly intuitive evolution editor built in now.  Making all the evolution changes manually is a bit tedious, but it only has to be done once

**Comment by CroixJeremy2** [comment]
Hello,
I also would like to change this option (or maybe make it available in a "tick box" in the randomizer at least), because I really would also like the default Marill... I checked where in the code you implemented this change. It is in:
com -> dabomstew -> pkrandom -> romhandlers -> Gen4RomHandler.java -> from line 4776 to line 4829.

I can delete those lines myself after having downloaded the universal-pokemon-randomizer-zx-master.zip file, but I have absolutely not any knowledge on how to make then the a new .jar file afterwards... I tried to download the latest .jar myself but I realised that it is now a .class file (not a .java that I edit myself). So, from what I have understood, a decompiler/editor tool is required. I tried myself with recaf-gui-0.5.3.jar, but when I try to save after having deleted the lines of interest, there are multiple errors elsewhere that I don't know how to solve... I think this is really a feature that only people that have already good knowledge of how to handle .java files and compilers can do unfortunately for us damned mortals. Could the dev change this feature please? or could anybody with those required skills tell us step by step how to do it please?

Thanks in advance for your response,
Best regards,

**Comment by ieris19** [comment]
@CroixJeremy2 You would need a Java environment. Check this guide out, it's pretty simple and official! https://github.com/Ajarmar/universal-pokemon-randomizer-zx/wiki/Building-Universal-Pokemon-Randomizer-ZX

**Comment by solbus** [comment]
This randomization seems to have been introduced in 4.2.0. If you use 4.1.0, it remains Marill.

**Comment by chacei** [comment]
Agreed, I don't think the partner pokemon should be affected - it breaks the immersion when all I really want is to be able to evolve an Alakazam. I'm using 4.6.1 and experiencing this so I'll try rolling back to 4.1.0.

**Comment by GitBreezy** [comment]
I guess I'm a +1 for wondering why this is intentionally coded into the randomizer. The way that I verified that it worked is by enabling running without running shoes. I was able to run, thus I knew I was good money. Either let people figure out on their own that the game is patched or let them choose another non-intrusive option lol

**Comment by southigator** [comment]
I also ran into this so added an option to disable this feature in PR #919. It adds a new option "Don't Randomize Inro Pokemon" to General Options

I have produced a new version if anyone would like to use it here:
https://github.com/southigator/universal-pokemon-randomizer-zx/releases/tag/v4.6.2
